File: Pythoncode/70.爬楼梯.py
class Solution:
    def climbStairs(self, n: int) -> int:
        d = {}
        return self.helper(n, d)
    def helper(self, n, d):
        if n in d:
            return d[n]
        if n < 0:
            return 0
        elif n == 0:
            return 1
        else:
            d[n] = self.helper(n-1, d) + self.helper(n-2, d)
            return d[n]

File: Cppcode/70.爬楼梯.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        vector<int> dp(n+1);
        for (auto i = 0; i < n+1; i++)
        {
            if (i == 0)
            {
                dp[i] = 1;
                continue;
            }
            auto a = i-2>=0 ? dp[i-2] : 0;
            auto b = i-1>=0 ? dp[i-1] : 0;
            dp[i] = a + b;
        }
        return dp[n];
    }
};